sympy/combinatorics/perm_groups.py

```diff
@@ -5075,3 +5075,132 @@ def _stabilizer(degree, generators, alpha):
+        """
+        if self._order is not None:
+            return self._order
+        n = self.deg
+        self._order = factorial(n)
+        return self._order
+
+    @property
+    def degree(self):
+        """
+        Return the degree of the permutation group.
+        """
+        return self.deg
+
+    @property
+    def identity(self):
+        '''
+        Return the identity element of the permutation group.
+
+        '''
+        return _af_new(list(range(self.deg)))
+
+class Coset(Basic):
+    """A left coset of a permutation group with respect to an element.
+
+    Parameters
+    ==========
+
+    g : Permutation
+
+    H : PermutationGroup
+
+    dir : "+" or "-", If not specified by default it will be "+"
+        here ``dir`` specified the type of coset "+" represent the
+        right coset and "-" represent the left coset.
+
+    G : PermutationGroup, optional
+        The group which contains *H* as its subgroup and *g* as its
+        element.
+
+        If not specified, it would automatically become a symmetric
+        group ``SymmetricPermutationGroup(g.size)`` and
+        ``SymmetricPermutationGroup(H.degree)`` if ``g.size`` and ``H.degree``
+        are matching.``SymmetricPermutationGroup`` is a lazy form of SymmetricGroup
+        used for representation purpose.
+    """
+
+    def __new__(cls, g, H, G=None, dir = "+"):
+        g = sympify(g)
+        if not isinstance(g, Permutation):
+            raise NotImplementedError
+
+        H = sympify(H)
+        if not isinstance(H, PermutationGroup):
+            raise NotImplementedError
+
+        if G is not None:
+            G = sympify(G)
+            if not isinstance(G, PermutationGroup):
+                raise NotImplementedError
+            if not H.is_subgroup(G):
+                raise ValueError("{} must be a subgroup of {}.".format(H, G))
+            if g not in G:
+                raise ValueError("{} must be an element of {}.".format(g, G))
+        else:
+            g_size = g.size
+            h_degree = H.degree
+            if g_size != h_degree:
+                raise ValueError(
+                    "The size of the permutation {} and the degree of "
+                    "the permutation group {} should be matching "
+                    .format(g, H))
+            G = SymmetricPermutationGroup(g.size)
+
+        dir = Symbol(dir)
+        if str(dir) ==  '+':
+            type = "RightCoset"
+        else:
+            type = "LeftCoset"
+        obj = Basic.__new__(cls, g, H, G, type)
+        obj._dir = dir
+        return obj
+
+    @property
+    def is_left_coset(self):
+        """
+        Check if the coset is left coset that is ``gH``.
+        """
+        return str(self._dir) == '-'
+
+    @property
+    def is_right_coset(self):
+        """
+        Check if the coset is right coset that is ``Hg``.
+        """
+        return str(self._dir) == '+'
+
+    def aslist(self):
```

Comment:
What is the convention that sympy is naming stuff that returns a list of its elements?
I think that this was from GAP was already in camelcase naming, so it should better be named like `as_list` to follow snakecase naming.